### PR TITLE
failpoint: add unit test for WithHook

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Add more unit test for failpoint.

### What is changed and how it works?

Add unit test for WithHook

### Check List 

Tests 

 - Unit test

Side effects

N/A

Related changes

N/A

### CODE REVIEW

**Comment by lonng** [comment]
PTAL @amyangfei

File: failpoint_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type failpointSuite struct{}
+
+func (s *failpointSuite) TestWithHook(c *C) {
+	ctx := failpoint.WithHook(context.Background(), func(ctx context.Context, fpname string) bool {
+		return false
+	})
+	ok, val := failpoint.Eval("unit-test", ctx)
+	c.Assert(ok, Equals, false)
+	c.Assert(val, Equals, nil)
```
Comment (kennytm) [inline failpoint_test.go]:
```suggestion
	c.Assert(val, IsNil)
```

File: failpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+var _ = &failpointSuite{}
+
+type failpointSuite struct{}
+
+func (s *failpointSuite) TestWithHook(c *C) {
+	ctx := failpoint.WithHook(context.Background(), func(ctx context.Context, fpname string) bool {
+		return false
+	})
+	ok, val := failpoint.Eval("unit-test", ctx)
+	c.Assert(ok, Equals, false)
```
Comment (kennytm) [inline failpoint_test.go]:
```suggestion
	c.Assert(ok, IsFalse)
```

File: failpoint_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	c.Assert(val, Equals, nil)
+
+	ctx = failpoint.WithHook(context.Background(), func(ctx context.Context, fpname string) bool {
+		return true
+	})
+	err := failpoint.Enable("TestWithHook-test-1", "return(1)")
+	defer func() {
+		err := failpoint.Disable("TestWithHook-test-1")
+		c.Assert(err, Equals, nil)
```
Comment (kennytm) [inline failpoint_test.go]:
```suggestion
		c.Assert(err, IsNil)
```

File: failpoint_test.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	}()
+	c.Assert(err, Equals, nil)
+	ok, val = failpoint.Eval("TestWithHook-test-1", ctx)
+	c.Assert(ok, Equals, true)
```
Comment (kennytm) [inline failpoint_test.go]:
```suggestion
	c.Assert(ok, IsTrue)
```

**Comment by lonng** [comment]
@kennytm PTAL again

File: failpoint_test.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+	ctx = failpoint.WithHook(context.Background(), func(ctx context.Context, fpname string) bool {
+		return true
+	})
+	err := failpoint.Enable("TestWithHook-test-1", "return(1)")
+	defer func() {
+		err := failpoint.Disable("TestWithHook-test-1")
+		c.Assert(err, IsNil)
+	}()
+	c.Assert(err, Equals, nil)
```
Comment (kennytm) [inline failpoint_test.go]: Also this 😅 

```suggestion
	c.Assert(err, IsNil)
```

**Pull request merged**